src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -106,12 +118,13 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             marshaller.marshal(toSave, fileWriter);
 
         } catch (IOException ioe) {
-            throw new StorageOperationException("Error writing to file: " + path);
+            throw new StorageOperationException("Error writing to file: " + path + " file might be read-only");
```

Comment:
Good job making the exception message more explicit to the user, but can `StorageOperationException` only be caused by a read-only error?